docs(cdk/portal): update portal docs (#19869)
angular/components: src/cdk/portal/portal.md

Usage:
```ts
ngAfterViewInit() {
  this.userSettingsPortal = new ComponentPortal(UserSettingsComponent);
}
```

##### `TemplatePortal`

> Used to create a portal from an embedded template.

Reviewer: ```suggestion
You can create a `TemplatePortal` from an `<ng-template>`. `TemplatePortal` allows you to take Angular content within one template and render it somewhere else.
```

Usage:
```html
<ng-template #templatePortalContent>Some content here</ng-template>
```

```ts

> @ViewChild('templatePortalContent') templatePortalContent: TemplateRef<any>;

Reviewer: ```suggestion
@ViewChild('templatePortalContent') templatePortalContent: TemplateRef<unknown>;
```

I'd want to use `unknown` here instead of `any` just to avoid showing examples that use `any` in general

ngAfterViewInit() {
  this.templatePortal = new TemplatePortal(
    this.templatePortalContent,
    this._viewContainerRef
  );
}
```

##### `DomPortal`

> Used to create a portal from a DOM element.

Reviewer: ```suggestion
You can create a `DomPortal` from any native DOM element. `DomPortal` allows you to take any arbitrary DOM content and render it somewhere else. `DomPortal` moves content _as is_, so elements with Angular features like bindings or directives may no longer update if moved via `DomPortal`.
```

Usage:
```html

> <div id="domPortalContent">Some content here</div>

Reviewer: ```suggestion
<div id="dom-portal-content">Some content here</div>
```
We try to use dash-case for IDs
